server/src/resolvers/getVariantsResolver/utils/fetchCaddAnnotations.ts

```diff
@@ -1,86 +1,91 @@
-import { exec } from 'child_process';
-import { promisify } from 'util';
 import resolveAssembly from './resolveAssembly';
+import resolveChromosome from './resolveChromosome';
 import { v4 as uuidv4 } from 'uuid';
 import { CADDAnnotationQueryResponse, CaddAnnotation } from '../../../types';
+import { TabixIndexedFile } from '@gmod/tabix';
+import { RemoteFile, Fetcher } from 'generic-filehandle';
+import fetch from 'node-fetch';
 
 const ANNOTATION_URL_38 =
   'https://krishna.gs.washington.edu/download/CADD/v1.6/GRCh38/whole_genome_SNVs_inclAnno.tsv.gz';
 const ANNOTATION_URL_37 =
-  'https://krishna.gs.washington.edu/download/CADD/v1.4/GRCh37/whole_genome_SNVs_inclAnno.tsv.gz';
+  'https://krishna.gs.washington.edu/download/CADD/v1.6/GRCh37/whole_genome_SNVs_inclAnno.tsv.gz';
 
 const INDEX_37_PATH = '/home/node/cadd_wgs_ghr37_index.gz.tbi';
 const INDEX_38_PATH = '/home/node/cadd_wgs_ghr38_index.gz.tbi';
 
-/*
-  indexes for headers in annotation tsv: 
-  https://cadd.gs.washington.edu/static/ReleaseNotes_CADD_v1.6.pdf
-  Chrom: 1
-  Pos: 2
-  Ref: 3
-  Alt: 4
-  Consequence: 8
-  oAA: 17
-  nAA: 18
-  FeatureID: 20
-  cDNApos: 25
-  protpos: 29
-*/
+const _getAnnotations = async (position: string, assemblyId: string) => {
+  const resolvedAssemblyId = resolveAssembly(assemblyId);
+  const annotationUrl = resolvedAssemblyId === '38' ? ANNOTATION_URL_38 : ANNOTATION_URL_37;
+  const indexPath = resolvedAssemblyId === '38' ? INDEX_38_PATH : INDEX_37_PATH;
 
-const NAME_MAP: Record<string, keyof CaddAnnotation> = {
-  Chr: 'chrom',
-  Pos: 'pos',
-  Ref: 'ref',
-  Alt: 'alt',
-  Consequence: 'consequence',
-  oAA: 'aaRef',
-  nAA: 'aaAlt',
-  FeatureID: 'transcript',
-  cDNApos: 'cdna',
-  protPos: 'aaPos',
-};
+  const tbiIndexed = new TabixIndexedFile({
+    filehandle: new RemoteFile(annotationUrl, { fetch: fetch as Fetcher }),
```

Comment:
Are you redownloading this dependency every time Tabix is used?

Reply:
The file is >300GB, but the query happens instantly, so no, I don't think the file specified with the `annotationUrl` is being downloaded every single time. I'm not entirely sure how tabix works under the hood, but according to the documentation [here](http://www.htslib.org/doc/tabix.html#:~:text=Last%20but%20not%20the%20least,without%20downloading%20the%20entire%20file.):

> Last but not the least, tabix supports remote data retrieval. One can put the data file and the index at an FTP or HTTP server, and other users or even web services will be able to get a slice without downloading the entire file.

Reply:
Okay, it's using HTTP byte-range headers to fetch chunks of it then.